sql/core/src/main/scala/org/apache/spark/sql/streaming/SnappyStreamingQueryListener.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Changes for SnappyData data platform.
+ *
+ * Portions Copyright (c) 2017-2019 TIBCO Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package org.apache.spark.sql.streaming
+
+import org.apache.spark.SparkContext
+
+class SnappyStreamingQueryListener(sparkContext: SparkContext) extends StreamingQueryListener {
+
+  val streamingRepo = StreamingRepository.getInstance
+
+  override def onQueryStarted(event: StreamingQueryListener.QueryStartedEvent): Unit = {
+    val queryName = {
+      if (event.name == null || event.name.isEmpty) {
+        event.id.toString
+      } else {
+        event.name
+      }
+    }
+
+    streamingRepo.allQueries.put(event.id,
```

Comment:
Dont see where we remove from allQueries, this could be memory leak in long running cluster?

Reply:
Done.
Also added spark configuration paramenter **_spark.sql.streaming.uiRunningQueriesDisplayLimit_** to configure how many maximum number of running queries be displayed on structure streaming UI.